### ISSUE TITLE
In JS, type annotation on require of TS export= isn't checked

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#26429]
**TypeScript Version:**  3.1.0-dev.20180810

**Code**

```ts
// @allowJs: true
// @checkJs: true
// @noEmit: true
// @strict: true
// @resolveJsonModule: true

// @Filename: /json.json
{ "a": 0 }

// @Filename: /js.js
module.exports = { a: 0 };

// @Filename: /user.js
const json0 = require("./json.json");
json0.b; // Error (good)

/** @type {{ b: number }} */
const json1 = require("./json.json"); // No error (bad)
json1.b; // No error (OK since that's the type annotation)

const js0 = require("./js.js");
json0.b; // Error (good)

/** @type {{ b: number }} */
const js1 = require("./js.js"); // Error (good)
js1.b;
```

**Expected behavior:**

Error at `const json1`, just like for `const js1`.

**Actual behavior:**

No error there.

**Comment by sheetalkamat** [comment]
If you import json file currently it behaves just like having export = expression in ts file, Ts file with export = { a: 10 } it does not report error as well. 
```typescript
// @allowJs: true
// @checkJs: true
// @noEmit: true
// @strict: true
// @resolveJsonModule: true

// @Filename: /ts.ts
export = { a: 0 };

// @Filename: /user.js
const ts0 = require("./ts");
ts0.b; // Error (good)

/** @type {{ b: number }} */
const ts1 = require("./ts"); // No error (bad)
ts1.b;
```

Adding @RyanCavanaugh and @DanielRosenwasser to check whats the correct behaviour here. For js module with `module.exports =` assignment, the expression type is widened and if its fresh object literal its regular type is used as apposed to type of expression of `export =`  of ts module. What should json module do?.

**Comment by RyanCavanaugh** [comment]
@sheetalkamat we should use the widened, non-fresh type (to avoid excess property errors)
